packages/coinstac-api-server/src/data/resolvers.js

```diff
@@ -641,34 +638,20 @@ const resolvers = {
         }
       }
 
+      const connection = await helperFunctions.getRethinkConnection();
+      const result = await rethink.table('pipelines')
+        .insert(args.pipeline, {
+          conflict: 'update',
+          returnChanges: true,
+        })
+        .run(connection);
 
-      return helperFunctions.getRethinkConnection()
-        .then((connection) =>
-          rethink.table('pipelines').insert(
-            args.pipeline,
-            {
-              conflict: "update",
-              returnChanges: true,
-            }
-          )
-          .run(connection)
-          .then((result) => rethink.table('pipelines')
-          .get(result.changes[0].new_val.id)
-          // Populate computations subfield with computation meta information
-          .merge(pipeline =>
```

Comment:
whats up with removing this?

Reply:
duplicate code.
check `fetchOnePipeline` function, @rssk

Reply:
jk, fetchpipe does the same